### PR TITLE
Add a11y to QueryViewer, DownloadData, BreadCrumbs, ECharts, DataTable Components

### PR DESCRIPTION
**Pull request by yukseltron**
### Description
Part of #489 

Most of these changes involved replacing `spans` with `button` tags and then styling them to match as they were before.

### DownloadData
Turned into a button for a11y.
Made `downloadData` function a prop w/ default.
Hover effect for DownloadData now through CSS in the parent component.
Updated ECharts and DataTable to both use DownloadData, both should still download the appropriate files for the user.

Before (ECharts):

https://user-images.githubusercontent.com/14843458/207218051-c8f5def8-5b56-44fd-a167-e1cb6bbc8f24.mov

After (ECharts):

https://user-images.githubusercontent.com/14843458/207218250-41123c2e-6bbf-42d7-8788-9ec207235f69.mov

Before (DataTable):

https://user-images.githubusercontent.com/14843458/207218701-9dc4aa0c-457f-4cff-b783-793ea3556103.mov

After (DataTable):

https://user-images.githubusercontent.com/14843458/207360872-9d3476c5-a1eb-4579-9fc6-13ca49c75012.mov

### BreadCrumbs
Updated 'Show/Hide Query' to be a11y and is now a button.

Before:

https://user-images.githubusercontent.com/14843458/207219412-733200af-fbc8-47ac-aef1-134a5fa09981.mov

After:

https://user-images.githubusercontent.com/14843458/207219521-726590e6-687d-4656-a92e-e2c42d5aad7a.mov

### QueryViewer
Updated to be a11y and is also now using buttons.

Before:

https://user-images.githubusercontent.com/14843458/207220000-ab9c934b-02e1-4163-b0bd-6ae0870b59a6.mov

After:

https://user-images.githubusercontent.com/14843458/207220121-8b0faec4-ebfb-403f-9242-fcd3fcac0e0c.mov

### CODE REVIEW

**Comment by archiewood** [comment]
I notice in the downloadData video, the styling of the numbers showing the rows seems to have changed in the Before vs After

**Comment by afzalsayed96** [comment]
@yukseltron In the interest of semantic correctness, shouldn't `type="button"` also be added to all the `button`s

**Comment by yukseltron** [comment]
@archiewood restored the styling for pagination.

@afzalsayed96 Yes! Added type='button' to the components.

**Review by archiewood — CHANGES_REQUESTED** [review]
Looks good overall.

Think the DowloadData is a sensible refactor

A few styling comments for you to look at

File: sites/example-project/src/components/ui/DownloadData.svelte [hunk 1]
```diff
@@ -39,42 +35,34 @@
...
 
-    .download-icon {
+    button {
         display: grid;
         grid-row: auto;
         grid-template-columns: auto auto;
         gap: 3px;
         float: right;
         /* margin-left: 5px; */
-        margin-top: 3px;
```
[the remark this answers is not in the document]
Comment (yukseltron) [reply sites/example-project/src/components/ui/DownloadData.svelte:49]: Added some margin to the top `10px`.
Screen Shot 2022-12-13 at 2 45 29 PM

File: sites/example-project/src/components/ui/BreadCrumbs.svelte [hunk 2]
```diff
@@ -59,17 +59,17 @@
         {#if $pageHasQueries}
             <span transition:blur|local>
                 {#if $showQueries}
-                <span class="dev-controls hide" on:click={toggleQueries}>Hide Queries</span>
+                <button type="button" class="dev-controls hide" on:click={toggleQueries}>Hide Queries</button>
```
[the remark this answers is not in the document]
Comment (yukseltron) [reply sites/example-project/src/components/ui/BreadCrumbs.svelte:62]: Hide Queries color fixed.


https://user-images.githubusercontent.com/14843458/207429876-37e37771-4847-4ee5-93f9-08d7a361b220.mov

**Review by archiewood — APPROVED** [review]

**Comment by mcrascal** [comment]
Epic, thank you @yukseltron.

**Pull request merged**